### PR TITLE
Z Moves

### PR DESCRIPTION
**Pull request by ghoulslash**
## Description
Adds z move engine including names, animations, effects, etc.

Thanks to AsparagusEduardo for the data collection/icon!

Also Handles:
- Trainer AI
zmove

- Multi Battles
zmove_multi

### CODE REVIEW

File: src/data/text/move_names.h [hunk 1]
```diff
@@ -749,3 +749,78 @@ const u8 gMoveNames[MOVES_COUNT][MOVE_NAME_LENGTH + 1] =
...
+static const u8 sText_10000000_Volt_Thunderbolt[] = _("10000000 Volt Thunderbolt");
+static const u8 sText_Stoked_Sparksurfer[] = _("Stoked Sparksurfer");
+static const u8 sText_Extreme_Evoboost[] = _("Extreme Evoboost");
+static const u8 sText_Pulverizing_Pancake[] = _("Pulverizing Pancake");
+static const u8 sText_Genesis_Supernova[] = _("Genesis Supernova");
+static const u8 sText_Sinister_Arrow_Raid[] = _("Sinister Arrow Raid");
+static const u8 sText_Malicious_Moonsault[] = _("Malicious Moonsault");
+static const u8 sText_Oceanic_Operetta[] = _("Oceanic Operetta");
+static const u8 sText_Splintered_Stormshards[] = _("Splintered Stormshards");
+static const u8 sText_Lets_Snuggle_Forever[] = _("Lets Snuggle Forever");
```
Comment (ExpoSeed) [inline src/data/text/move_names.h:781]: `'`

Comment (ExpoSeed) [reply src/data/text/move_names.h:781]:
```suggestion
static const u8 sText_Lets_Snuggle_Forever[] = _("Let's Snuggle Forever");
```

**Comment by AsparagusEduardo** [comment]
We need a new icon for the trigger, the current one I made as a joke/placeholder xD

File: src/data/battle_moves.h [hunk 2]
```diff
@@ -11386,4 +12720,533 @@ const struct BattleMove gBattleMoves[MOVES_COUNT] =
...
+    {
+        .effect = EFFECT_ALL_STATS_UP_HIT,
+        .power = 185,
+        .type = TYPE_DRAGON,
+        .accuracy = 0,
+        .pp = 1,
+        .secondaryEffectChance = 100,
+        .target = MOVE_TARGET_BOTH,
+        .priority = 0,
+        .flags = 0,
```
Comment (LOuroboros) [inline src/data/battle_moves.h:13172]: This Z-Mve move is affected by Soundproof, so it should have the `FLAG_SOUND` set.

**Comment by Deidara9216** [comment]
Hello, could you modify the z-moves so that they are binary entered, it is for an old rom in which I have spent a lot of time.

**Comment by ExpoSeed** [comment]
Could you clarify "binary entered"?

This particular PR is mostly additive and does not remove much, so it should not cause major conflicts if you're pulling into an old repo. If you are, you'll have problems elsewhere, not with z moves.

**Comment by Deidara9216** [comment]
I mean the ASM method could you please adapt it, it is that I don't want to start over from scratch.

**Comment by LOuroboros** [comment]
> 
> 
> I mean the ASM method could you please adapt it, it is that I don't want to start over from scratch.

This was not written in ASM, this was quite clearly written in C, the same programming language in which the GBA Games and the decomps are written in.

You seem to be asking for someone in here to write an implementation of Z-Moves in ASM and inject it into your ROM, is that correct?

That is more than likely not going to happen, but in any case, this is simply not the right place to ask for something like that.

**Comment by ExpoSeed** [comment]
would you mind syncing this again? there are conflicts

**Comment by ghoulslash** [comment]
> 
> 
> would you mind syncing this again? there are conflicts

done

File: src/battle_anim_new.c [hunk 3]
```diff
@@ -4948,7 +7552,193 @@ static void SpriteCB_Geyser(struct Sprite* sprite)
...
+	/* TODO - lycanroc forms
+    #ifdef SPECIES_LYCANROC_MIDNIGHT
+	if (GetIllusionPartyData(gBattleAnimAttacker)->species == SPECIES_LYCANROC_MIDNIGHT)
+		gBattleAnimArgs[0] = 1;
+	else
+	#endif
+		gBattleAnimArgs[0] = 0;*/
+    
+    gBattleAnimArgs[0] = 0;
+	DestroyAnimVisualTask(taskId);
```
Comment (ExpoSeed) [inline src/battle_anim_new.c:7698]: this entire file mixes tabs and spaces, please change them all to spaces

File: src/battle_anim_new.c [hunk 4]
```diff
@@ -4948,7 +7552,193 @@ static void SpriteCB_Geyser(struct Sprite* sprite)
...
+
+void AnimTask_GetLycanrocForm(u8 taskId)
+{
+	/* TODO - lycanroc forms
+    #ifdef SPECIES_LYCANROC_MIDNIGHT
+	if (GetIllusionPartyData(gBattleAnimAttacker)->species == SPECIES_LYCANROC_MIDNIGHT)
+		gBattleAnimArgs[0] = 1;
+	else
+	#endif
+		gBattleAnimArgs[0] = 0;*/
```
Comment (ExpoSeed) [inline src/battle_anim_new.c:7695]: is there a reason why this is todo? i'm pretty sure PE has midnight lycanroc so there shouldn't be an issue with this check

File: src/battle_anim_new.c [hunk 5]
```diff
@@ -4149,7 +6627,133 @@ static void InitSpritePosToAnimAttackersCentre(struct Sprite *sprite, bool8 resp
     sprite->pos1.y += gBattleAnimArgs[1];
 }
 
-//sprite callbacks
+////sprite callbacks
```
Comment (ExpoSeed) [inline src/battle_anim_new.c:6630]: lol
```suggestion
//sprite callbacks
```

File: src/battle_anim_new.c [hunk 6]
```diff
@@ -4062,8 +4078,2470 @@ const struct SpriteTemplate gSteelBeamSpikeShardTemplate =
...
+    .oam = &gOamData_AffineOff_ObjNormal_16x16,
+    .anims = gDummySpriteAnimTable,
+    .images = NULL,
+    .affineAnims = gDummySpriteAffineAnimTable,
+    .callback = AnimMudSportDirt
+};
+
+// seering sunraze smash
+static const union AffineAnimCmd sSearingSunrazeSmashWormholeGrowingAffineAnims[] = {
+    //.hword 0x100,0x100,0x0,0x0, 0x0,0x0,0x88f6,0x0, 0x7fff,0x0,0x0,0x0
```
Comment (ExpoSeed) [inline src/battle_anim_new.c:6153]: are these comments necessary? if not, remove all instances of `//.hword` comments

File: src/battle_z_move.c [hunk 7]
```diff
@@ -0,0 +1,722 @@
...
+    {
+        // clear move slots
+        for (i = 0; i < MAX_MON_MOVES; ++i)
+        {
+            MoveSelectionDestroyCursorAt(i);
+            StringCopy(gDisplayedStringBattle, gText_EmptyString2);
+            BattlePutTextOnWindow(gDisplayedStringBattle, i + 3); 
+        }
+
+        //if (zmove == MOVE_Z_STATUS)
```
Comment (ExpoSeed) [inline src/battle_z_move.c:414]:
```suggestion
```

**Review by ExpoSeed — CHANGES_REQUESTED** [review]
logic of the code seems to make sense, just some minor style nitpicks that can be taken care of alongside solving the conflicts

**Comment by ExpoSeed** [comment]
a good chunk of them are probably leftover comments from cfru which don't apply to emerald

File: src/battle_z_move.c [hunk 8]
```diff
@@ -0,0 +1,720 @@
...
+        return FALSE;   //cant use z move twice
+    
+    if (IsViableZMove(battlerAtk, chosenMove))
+    {
+        #ifdef POKEMON_EXPANSION
+        if (defAbility == ABILITY_DISGUISE && defSpecies == SPECIES_MIMIKYU)
+            return 0; // Don't waste a Z-Move busting Mimikyu's disguise
+        if (defAbility == ABILITY_ICEFACE && defSpecies == SPECIES_EISCUE && IS_MOVE_PHYSICAL(chosenMove))
+            return 0; // Don't waste a Z-Move busting Eiscue's Ice Face
+        #endif
```
Comment (ExpoSeed) [inline src/battle_z_move.c:706]:
```suggestion
        #ifdef POKEMON_EXPANSION
        if (gBattleMons[battlerAtk].ability == ABILITY_DISGUISE && gBattleMons[battlerDef].species == SPECIES_MIMIKYU)
            return 0; // Don't waste a Z-Move busting Mimikyu's disguise
        if (gBattleMons[battlerAtk].ability == ABILITY_ICE_FACE && gBattleMons[battlerDef].species == SPECIES_EISCUE && IS_MOVE_PHYSICAL(chosenMove))
            return 0; // Don't waste a Z-Move busting Eiscue's Ice Face
        #endif
```

**Review by ExpoSeed — CHANGES_REQUESTED** [review]
fix conflicts as well (even though i only got one conflict so ¯\_(ツ)_/¯)

apologies for delaying on reviewing this, i had winter storm and general demotivation

i'm reviewing the thing in game as well at the moment

**Comment by AsparagusEduardo** [comment]
Doing some testing, Z-move animations are currently shifted +1 (Breakneck Blitz plays All-Out Pummeling's animation and so forth).

File: src/battle_z_move.c [hunk 9]
```diff
@@ -0,0 +1,721 @@
...
+    {SPECIES_MIMIKYU,               ITEM_MIMIKIUM_Z,           MOVE_PLAY_ROUGH,          MOVE_LETS_SNUGGLE_FOREVER},
+    {SPECIES_MIMIKYU_BUSTED,        ITEM_MIMIKIUM_Z,           MOVE_PLAY_ROUGH,          MOVE_LETS_SNUGGLE_FOREVER},
+    {SPECIES_PIKACHU_ORIGINAL_CAP,  ITEM_PIKASHUNIUM_Z,        MOVE_THUNDERBOLT,         MOVE_10000000_VOLT_THUNDERBOLT},
+    {SPECIES_PIKACHU_HOENN_CAP,     ITEM_PIKASHUNIUM_Z,        MOVE_THUNDERBOLT,         MOVE_10000000_VOLT_THUNDERBOLT},
+    {SPECIES_PIKACHU_SINNOH_CAP,    ITEM_PIKASHUNIUM_Z,        MOVE_THUNDERBOLT,         MOVE_10000000_VOLT_THUNDERBOLT},
+    {SPECIES_PIKACHU_UNOVA_CAP,     ITEM_PIKASHUNIUM_Z,        MOVE_THUNDERBOLT,         MOVE_10000000_VOLT_THUNDERBOLT},
+    {SPECIES_PIKACHU_KALOS_CAP,     ITEM_PIKASHUNIUM_Z,        MOVE_THUNDERBOLT,         MOVE_10000000_VOLT_THUNDERBOLT},
+    {SPECIES_PIKACHU_ALOLA_CAP,     ITEM_PIKASHUNIUM_Z,        MOVE_THUNDERBOLT,         MOVE_10000000_VOLT_THUNDERBOLT},
+    {SPECIES_PIKACHU_PARTNER_CAP,   ITEM_PIKASHUNIUM_Z,        MOVE_THUNDERBOLT,         MOVE_10000000_VOLT_THUNDERBOLT},
```
Comment (ExpoSeed) [inline src/battle_z_move.c:80]: world cap pikachu should be able to use 10m volt tbolt

File: src/battle_z_move.c [hunk 10]
```diff
@@ -0,0 +1,721 @@
...
+    {SPECIES_SOLGALEO,              ITEM_SOLGANIUM_Z,          MOVE_SUNSTEEL_STRIKE,     MOVE_SEARING_SUNRAZE_SMASH},
+    {SPECIES_NECROZMA_DUSK_MANE,    ITEM_SOLGANIUM_Z,          MOVE_SUNSTEEL_STRIKE,     MOVE_SEARING_SUNRAZE_SMASH},
+    {SPECIES_TAPU_KOKO,             ITEM_TAPUNIUM_Z,           MOVE_NATURES_MADNESS,     MOVE_GUARDIAN_OF_ALOLA},
+    {SPECIES_TAPU_BULU,             ITEM_TAPUNIUM_Z,           MOVE_NATURES_MADNESS,     MOVE_GUARDIAN_OF_ALOLA},
+    {SPECIES_TAPU_LELE,             ITEM_TAPUNIUM_Z,           MOVE_NATURES_MADNESS,     MOVE_GUARDIAN_OF_ALOLA},
+    {SPECIES_TAPU_FINI,             ITEM_TAPUNIUM_Z,           MOVE_NATURES_MADNESS,     MOVE_GUARDIAN_OF_ALOLA},
+    {SPECIES_NECROZMA_ULTRA,        ITEM_ULTRANECROZIUM_Z,     MOVE_PHOTON_GEYSER,       MOVE_LIGHT_THAT_BURNS_THE_SKY},
+    #endif
+    {SPECIES_MEW,                   ITEM_MEWNIUM_Z,            MOVE_PSYCHIC,             MOVE_GENESIS_SUPERNOVA},
+    {SPECIES_PIKACHU,               ITEM_PIKANIUM_Z,           MOVE_VOLT_TACKLE,         MOVE_CATASTROPIKA},
```
Comment (ExpoSeed) [inline src/battle_z_move.c:91]: i think cosplay pikachus should be able to use catastropika

**Review by ExpoSeed — CHANGES_REQUESTED** [review]

File: src/data/battle_moves.h [hunk 11]
```diff
@@ -9688,8 +10929,10 @@ const struct BattleMove gBattleMoves[MOVES_COUNT] =
         .secondaryEffectChance = 0,
         .target = MOVE_TARGET_SELECTED,
         .priority = 0,
-        .flags = FLAG_MAKES_CONTACT | FLAG_PROTECT_AFFECTED | FLAG_MIRROR_MOVE_AFFECTED | FLAG_IRON_FIST_BOOST,
+        .flags = FLAG_MAKES_CONTACT | FLAG_PROTECT_AFFECTED | FLAG_MIRROR_MOVE_AFFECTED,
```
Comment (amiosi) [inline src/data/battle_moves.h:10932]: should also be boosted by iron fist

File: src/data/battle_moves.h [hunk 12]
```diff
@@ -9774,12 +11027,14 @@ const struct BattleMove gBattleMoves[MOVES_COUNT] =
...
         .split = SPLIT_STATUS,
+        .zMovePower = 0,
+        .zMoveEffect = Z_EFFECT_SPDEF_UP_1,
     },
 
     [MOVE_TOXIC_THREAD] =
     {
         .effect = EFFECT_TOXIC_THREAD,
-        .power = 0,
+        .power = 20,
```
Comment (amiosi) [inline src/data/battle_moves.h:11037]: power = 0, pp = 20

File: src/data/battle_moves.h [hunk 13]
```diff
@@ -9577,13 +10804,15 @@ const struct BattleMove gBattleMoves[MOVES_COUNT] =
...
+        .zMovePower = 190,
+        .zMoveEffect = Z_EFFECT_NONE,
     },
 
     [MOVE_HYPERSPACE_FURY] =
     {
         .effect = EFFECT_PLACEHOLDER,
         .power = 0,
-        .type = TYPE_DARK,
+        .type = TYPE_MYSTERY,
```
Comment (amiosi) [inline src/data/battle_moves.h:10815]: dark type

File: src/data/battle_moves.h [hunk 14]
```diff
@@ -842,8 +946,8 @@ const struct BattleMove gBattleMoves[MOVES_COUNT] =
         .secondaryEffectChance = 10,
         .target = MOVE_TARGET_SELECTED,
         .priority = 0,
-        .flags = FLAG_PROTECT_AFFECTED | FLAG_MIRROR_MOVE_AFFECTED | FLAG_SHEER_FORCE_BOOST,
-        .split = SPLIT_SPECIAL,
```
Comment (amiosi) [inline src/data/battle_moves.h:846]: mistaken removal

File: src/data/battle_moves.h [hunk 15]
```diff
@@ -1572,9 +1769,11 @@ const struct BattleMove gBattleMoves[MOVES_COUNT] =
         .pp = 20,
         .secondaryEffectChance = 0,
         .target = MOVE_TARGET_USER,
-        .priority = -6,
+        .priority = 0,
```
Comment (amiosi) [inline src/data/battle_moves.h:1772]: priority = -6

File: src/data/battle_moves.h [hunk 16]
```diff
@@ -1589,20 +1788,24 @@ const struct BattleMove gBattleMoves[MOVES_COUNT] =
...
+        .zMoveEffect = Z_EFFECT_NONE,
     },
 
     [MOVE_MIMIC] =
     {
         .effect = EFFECT_MIMIC,
         .power = 0,
         .type = TYPE_NORMAL,
-        .accuracy = 0,
+        .accuracy = 100,
```
Comment (amiosi) [inline src/data/battle_moves.h:1800]: accuracy = 0

File: src/data/battle_moves.h [hunk 17]
```diff
@@ -2096,8 +2361,8 @@ const struct BattleMove gBattleMoves[MOVES_COUNT] =
         .pp = 15,
         .secondaryEffectChance = 0,
         .target = MOVE_TARGET_SELECTED,
-        .priority = 0,
-        .split = SPLIT_STATUS,
```
Comment (amiosi) [inline src/data/battle_moves.h:2100]: mistaken removals

File: src/data/battle_moves.h [hunk 18]
```diff
@@ -2869,9 +3228,11 @@ const struct BattleMove gBattleMoves[MOVES_COUNT] =
         .pp = 10,
         .secondaryEffectChance = 0,
         .target = MOVE_TARGET_USER,
-        .priority = 4,
+        .priority = 3,
```
Comment (amiosi) [inline src/data/battle_moves.h:3231]: priority = 4

File: src/data/battle_moves.h [hunk 19]
```diff
@@ -3107,9 +3496,11 @@ const struct BattleMove gBattleMoves[MOVES_COUNT] =
         .pp = 5,
         .secondaryEffectChance = 0,
         .target = MOVE_TARGET_USER,
-        .priority = 4,
+        .priority = 3,
```
Comment (amiosi) [inline src/data/battle_moves.h:3499]: priority = 4

File: src/data/battle_moves.h [hunk 20]
```diff
@@ -3215,9 +3616,11 @@ const struct BattleMove gBattleMoves[MOVES_COUNT] =
         .pp = 10,
         .secondaryEffectChance = 0,
         .target = MOVE_TARGET_USER,
-        .priority = 4,
+        .priority = 3,
```
Comment (amiosi) [inline src/data/battle_moves.h:3619]: priority = 4

File: src/data/battle_moves.h [hunk 21]
```diff
@@ -3487,7 +3924,9 @@ const struct BattleMove gBattleMoves[MOVES_COUNT] =
         .target = MOVE_TARGET_SELECTED,
         .priority = 0,
         .flags = FLAG_PROTECT_AFFECTED | FLAG_MIRROR_MOVE_AFFECTED | FLAG_SHEER_FORCE_BOOST,
-        .split = SPLIT_PHYSICAL,
+        .split = SPLIT_SPECIAL,
```
Comment (amiosi) [inline src/data/battle_moves.h:3927]: SPLIT_PHYSICAL

File: src/data/battle_moves.h [hunk 22]
```diff
@@ -3650,20 +4109,24 @@ const struct BattleMove gBattleMoves[MOVES_COUNT] =
...
+        .zMoveEffect = Z_EFFECT_NONE,
     },
 
     [MOVE_VITAL_THROW] =
     {
         .effect = EFFECT_VITAL_THROW,
         .power = 70,
         .type = TYPE_FIGHTING,
-        .accuracy = 0,
+        .accuracy = 100,
```
Comment (amiosi) [inline src/data/battle_moves.h:4121]: accuracy = 0

File: src/data/battle_moves.h [hunk 23]
```diff
@@ -3764,8 +4239,10 @@ const struct BattleMove gBattleMoves[MOVES_COUNT] =
         .secondaryEffectChance = 0,
         .target = MOVE_TARGET_USER,
         .priority = 0,
-        .flags = 0,
+        .flags = FLAG_DANCE,
```
Comment (amiosi) [inline src/data/battle_moves.h:4242]: .flags = 0,

File: src/data/battle_moves.h [hunk 24]
```diff
@@ -4203,9 +4730,11 @@ const struct BattleMove gBattleMoves[MOVES_COUNT] =
         .pp = 20,
         .secondaryEffectChance = 0,
         .target = MOVE_TARGET_USER,
-        .priority = 2,
+        .priority = 3,
```
Comment (amiosi) [inline src/data/battle_moves.h:4733]: priority = 2

File: src/data/battle_moves.h [hunk 25]
```diff
@@ -4250,8 +4783,8 @@ const struct BattleMove gBattleMoves[MOVES_COUNT] =
         .pp = 20,
         .secondaryEffectChance = 0,
         .target = MOVE_TARGET_SELECTED,
-        .priority = 0,
-        .split = SPLIT_STATUS,
```
Comment (amiosi) [inline src/data/battle_moves.h:4254]: mistaken removal

File: src/data/battle_moves.h [hunk 26]
```diff
@@ -4284,20 +4819,24 @@ const struct BattleMove gBattleMoves[MOVES_COUNT] =
...
+        .zMoveEffect = Z_EFFECT_SPD_UP_2,
     },
 
     [MOVE_ROLE_PLAY] =
     {
         .effect = EFFECT_ROLE_PLAY,
         .power = 0,
         .type = TYPE_PSYCHIC,
-        .accuracy = 0,
+        .accuracy = 100,
```
Comment (amiosi) [inline src/data/battle_moves.h:4831]: accuracy = 0

File: src/data/battle_moves.h [hunk 27]
```diff
@@ -4314,8 +4853,8 @@ const struct BattleMove gBattleMoves[MOVES_COUNT] =
         .pp = 10,
         .secondaryEffectChance = 0,
         .target = MOVE_TARGET_USER,
-        .priority = 0,
-        .split = SPLIT_STATUS,
```
Comment (amiosi) [inline src/data/battle_moves.h:4318]: mistaken removal

File: src/data/battle_moves.h [hunk 28]
```diff
@@ -4418,20 +4969,24 @@ const struct BattleMove gBattleMoves[MOVES_COUNT] =
...
+        .zMoveEffect = Z_EFFECT_NONE,
     },
 
     [MOVE_YAWN] =
     {
         .effect = EFFECT_YAWN,
         .power = 0,
         .type = TYPE_NORMAL,
-        .accuracy = 0,
+        .accuracy = 100,
```
Comment (amiosi) [inline src/data/battle_moves.h:4981]: accuracy = 0

File: src/data/battle_moves.h [hunk 29]
```diff
@@ -4478,20 +5037,24 @@ const struct BattleMove gBattleMoves[MOVES_COUNT] =
...
+        .zMoveEffect = Z_EFFECT_NONE,
     },
 
     [MOVE_SKILL_SWAP] =
     {
         .effect = EFFECT_SKILL_SWAP,
         .power = 0,
         .type = TYPE_PSYCHIC,
-        .accuracy = 0,
+        .accuracy = 100,
```
Comment (amiosi) [inline src/data/battle_moves.h:5049]: accuracy = 0

File: src/data/battle_moves.h [hunk 30]
```diff
@@ -5285,8 +5948,10 @@ const struct BattleMove gBattleMoves[MOVES_COUNT] =
         .secondaryEffectChance = 0,
         .target = MOVE_TARGET_USER,
         .priority = 0,
-        .flags = FLAG_SNATCH_AFFECTED | FLAG_SOUND,
+        .flags = FLAG_SNATCH_AFFECTED,
```
Comment (amiosi) [inline src/data/battle_moves.h:5951]: should also have sound flag

File: src/data/battle_moves.h [hunk 31]
```diff
@@ -7022,11 +7904,13 @@ const struct BattleMove gBattleMoves[MOVES_COUNT] =
...
         .flags = FLAG_PROTECT_AFFECTED | FLAG_SOUND | FLAG_SHEER_FORCE_BOOST,
         .split = SPLIT_SPECIAL,
+        .zMovePower = 120,
+        .zMoveEffect = Z_EFFECT_NONE,
     },
 
     [MOVE_JUDGMENT] =
     {
-        .effect = EFFECT_CHANGE_TYPE_ON_ITEM,
+        .effect = EFFECT_HIT,
```
Comment (amiosi) [inline src/data/battle_moves.h:7913]: regression?

File: src/data/battle_moves.h [hunk 32]
```diff
@@ -7036,7 +7920,8 @@ const struct BattleMove gBattleMoves[MOVES_COUNT] =
         .priority = 0,
         .flags = FLAG_PROTECT_AFFECTED | FLAG_MIRROR_MOVE_AFFECTED | FLAG_KINGS_ROCK_AFFECTED,
         .split = SPLIT_SPECIAL,
-        .argument = HOLD_EFFECT_PLATE,
```
Comment (amiosi) [inline src/data/battle_moves.h:7039]: regression here too

File: src/data/battle_moves.h [hunk 33]
```diff
@@ -7412,11 +8345,13 @@ const struct BattleMove gBattleMoves[MOVES_COUNT] =
...
         .flags = FLAG_PROTECT_AFFECTED | FLAG_MIRROR_MOVE_AFFECTED | FLAG_KINGS_ROCK_AFFECTED,
         .split = SPLIT_SPECIAL,
+        .zMovePower = 120,
+        .zMoveEffect = Z_EFFECT_NONE,
     },
 
     [MOVE_AUTOTOMIZE] =
     {
-        .effect = EFFECT_AUTOTOMIZE,
+        .effect = EFFECT_HIT,
```
Comment (amiosi) [inline src/data/battle_moves.h:8354]: effect_autotomize

File: src/data/battle_moves.h [hunk 34]
```diff
@@ -10038,8 +11329,10 @@ const struct BattleMove gBattleMoves[MOVES_COUNT] =
         .secondaryEffectChance = 0,
         .target = MOVE_TARGET_SELECTED,
         .priority = -3,
-        .flags = FLAG_PROTECT_AFFECTED | FLAG_BALLISTIC,
+        .flags = FLAG_PROTECT_AFFECTED,
```
Comment (amiosi) [inline src/data/battle_moves.h:11332]: should also have ballistic flag

File: src/data/battle_moves.h [hunk 35]
```diff
@@ -10624,9 +11961,11 @@ const struct BattleMove gBattleMoves[MOVES_COUNT] =
         .accuracy = 100,
         .pp = 5,
         .secondaryEffectChance = 30,
-        .target = MOVE_TARGET_SELECTED,
+        .target = MOVE_TARGET_FOES_AND_ALLY,
```
Comment (amiosi) [inline src/data/battle_moves.h:11964]: target selected

**Review by amiosi — CHANGES_REQUESTED** [review]

**Comment by ProjectRevoTPP** [comment]
https://github.com/rh-hideout/pokeemerald-expansion/blob/71b85031e0dee8c91d039fd7075c169070b594ba/src/battle_util.c#L8303

This is a NULL deference issue. Make sure you check if the PTR is there, otherwise it will cause the wrong status icons to be displayed on the summary screen.

File: src/battle_anim_new.c [hunk 36]
```diff
@@ -4062,204 +4074,2788 @@ const struct SpriteTemplate gSteelBeamSpikeShardTemplate =
...
+    .oam = &gOamData_AffineNormal_ObjNormal_64x64,
+    .anims = gDummySpriteAnimTable,
+    .images = NULL,
+    .affineAnims = gAffineAnims_GrowingElectricOrb,
+    .callback = AnimGrowingChargeOrb
+};
+const struct SpriteTemplate gContinentalCrushEruptionSpriteTemplate = 
+{
+    .tileTag = ANIM_TAG_ROCKS,
+    .paletteTag ANIM_TAG_ROCKS,
```
Comment (ExpoSeed) [inline src/battle_anim_new.c:4261]:
```suggestion
    .paletteTag = ANIM_TAG_ROCKS,
```

File: src/battle_anim_new.c [hunk 37]
```diff
@@ -4062,204 +4074,2788 @@ const struct SpriteTemplate gSteelBeamSpikeShardTemplate =
...
+    .oam = &gOamData_AffineOff_ObjNormal_16x32,
+    .anims = gEndureEnergyAnimTable,
+    .images = NULL,
+    .affineAnims = gDummySpriteAffineAnimTable,
+    .callback = AnimEndureEnergy
+};
+const struct SpriteTemplate gNeverEndingNightmareHandSpriteTemplate = 
+{
+    .tileTag = ANIM_TAG_ASSURANCE_HAND,
+    .paletteTag ANIM_TAG_ASSURANCE_HAND,
```
Comment (ExpoSeed) [inline src/battle_anim_new.c:4355]:
```suggestion
    .paletteTag = ANIM_TAG_ASSURANCE_HAND,
```

File: src/battle_anim_new.c [hunk 38]
```diff
@@ -4062,204 +4074,2788 @@ const struct SpriteTemplate gSteelBeamSpikeShardTemplate =
...
+    .oam = &gOamData_AffineOff_ObjNormal_16x16,
+    .anims = gPetalDanceBigFlowerAnimTable,
+    .images = NULL,
+    .affineAnims = gDummySpriteAffineAnimTable,
+    .callback = AnimFireSpiralOutward
+};
+const struct SpriteTemplate gBloomDoomGreenChargeSpriteTemplate = 
+{
+    .tileTag = ANIM_TAG_CIRCLE_OF_LIGHT,
+    .paletteTag ANIM_TAG_RAZOR_LEAF,
```
Comment (ExpoSeed) [inline src/battle_anim_new.c:4603]:
```suggestion
    .paletteTag = ANIM_TAG_RAZOR_LEAF,
```

**Review by ExpoSeed — CHANGES_REQUESTED** [review]
in addition, pls merge with be and see the previous review comments i made about pikachu's z moves

**Comment by ExpoSeed** [comment]
superseded by #1543

**Pull request closed without merging**